src/test/java/rx/CompletableTest.java

```diff
@@ -357,6 +357,56 @@ public void call(Long v) {
         Assert.assertEquals(Arrays.asList(5L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L), requested);
     }
     
+    @Test
+    public void andThen() {
+        TestSubscriber<String> ts = new TestSubscriber<>(0);
+        Completable.complete().andThen(Observable.just("foo")).subscribe(ts);
+        ts.requestMore(1);
+        ts.assertValue("foo");
+        ts.assertCompleted();
+        ts.assertNoErrors();
+    }
+    
+    @Test
+    public void andThenNever() {
+        TestSubscriber<String> ts = new TestSubscriber<>(0);
+        Completable.never().andThen(Observable.just("foo")).subscribe(ts);
+        ts.requestMore(1);
+        ts.assertNoValues();
+        ts.assertNoTerminalEvent();
+    }
+    
+    @Test
+    public void andThenError() {
+        TestSubscriber<String> ts = new TestSubscriber<>(0);
+        AtomicBoolean hasRun = new AtomicBoolean(false);
+        Exception e = new Exception();
+        Completable.create(cs -> cs.onError(e))
```

Comment:
Lambdas are Java 8. Besides, there is already a `Completable.error()` and you should call `cs.onSubscribe(Subscriptions.unsubscribed())` first.